htdocs/main.inc.php

```diff
@@ -36,6 +36,21 @@
  *	\brief      File that defines environment for Dolibarr GUI pages only (file not required by scripts)
  */
 
+// Log WARNING to trace_warningphp.log and not display it to user
+error_reporting(E_ALL ^ E_NOTICE);
+@set_error_handler("__error_handler");
+function __error_handler($errno, $errstr, $errfile, $errline)
+{
+	//echo "<pre><br/>ERREUR/WARNING : $errno, $errstr, $errfile, $errline<br/></pre>"; // for display it to user
+	trace("ERREUR/WARNING: $errno, $errstr, $errfile, $errline");
+}
+function trace($toTrace)
+{
+	$f = fopen("trace_warningphp.log", "a+");
```

Comment:
On a secured installation, the application has no permission to write into a file into htdocs directory, so this won't work.

Also, if creating log, rotation of log must be managed.

Also hidding errors to users with no choice of user is not a good practice. user must choose if he want to not see important information he should see.

You don't need this to report errors onto a log file. You can already use the module "Log debug" but also the module "Debug bar".